modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqexecEngine.scala

```diff
@@ -210,6 +211,8 @@ class SeqexecEngine(settings: SeqexecEngine.Settings) {
       // TODO: Calculate the whole status here and remove `Engine.Step.status`
       // This will be easier once the exact status labels in the UI are fixed.
       seq.steps.map(viewStep) match {
+        // The sequence could be empty
+        case Nil => Nil
         // Find first Pending Step when no Step is Running and mark it as Running
         case steps if (st === SequenceState.Running || st === SequenceState.Stopping) && steps.all(_.status =/= StepState.Running) =>
           val (xs, (y :: ys)) = splitWhere(steps)(_.status === StepState.Pending)
```

Comment:
This is what we're using in `gem` and I think it is helping to keep the code cleaner:

https://github.com/gemini-hlsw/gem/blob/master/build.sbt#L83-L134